### DISCUSSION TITLE
[webpack] [swc-loader] Unable to open actual source files with generated Source Maps

### DISCUSSION THREAD

**Discussion by phoenisx** [swc-project/swc#2252] [Q&A]
Hi

I am not sure if this is a webpack/swc issue, because source maps look fine, but I am not able to open the actual source file mapped with the transpiled code.

_I am trying to set up webpack with swc-loader, everything's working fine. Even the sourceMappingURL is pointing to the correct location (at least relatively)._

`.js.map` contains redirect URL to actual source to something like `webpack:///src/...` for webpack 4.x and `webpack://{roodir}/src/...` which I feel is correct (not sure though). 

The problem is both the above source URLs are not opening, and I am getting the following error:

```sh
Error while fetching an original source: NetworkError when attempting to fetch resource.
Source URL: webpack://src/...
```

Webpack config has:
```
devtool: "source-map",
module: {
    rules: [
      {
        test: /\.m?jsx?$/,
        exclude: /node_modules/,
        use: [
          {
            loader: "swc-loader",
            options: swcJSConfig
          }
        ]
      },
      {
        test: /\.tsx?$/,
        exclude: /node_modules/,
        use: [
          {
            loader: "swc-loader",
            options: swcTSConfig
          }
        ]
      },
    ]
}
```

where swcConfig is similar for both ts/js templates:
```js
const CONFIG = {
  sourceMaps: true,
  test: ".m?jsx?$",
  jsc: {
    target: "es2016",
    loose: true,
    parser: {
      syntax: "ecmascript",
      jsx: true,
      dynamicImport: true
    },
    transform: {
      react: {
        runtime: "automatic",
        refresh: isDev,
      }
    }
  }
};

export default CONFIG;
```

The same issue happens with official swc examples as well https://github.com/swc-project/swc-loader/tree/master/example

Please help me out here, as I am trying to push `swc` tooling to our production, and without source mapping it would be hard to convince the team. 🙏🏾 🙇🏾

**Comment by phoenisx** [comment]
To add more details for the above, attaching a screenshot which shows the issue with  https://github.com/swc-project/swc-loader/tree/master/example as well

<img width="1477" alt="image" src="https://user-images.githubusercontent.com/11786283/133319084-dadea837-f217-4ee2-871e-241d627970f1.png">

**Comment by phoenisx** [comment]
I found this is happening due to `swc` / `swc-loader`, because replacing the loader with `babel-loader` for the example project mentioned above, made the source map work.

**Reply by phoenisx** [reply]
Adding link to `webpack` discussion for the same: https://github.com/webpack/webpack/discussions/14252.

**_This looks like a `@swc/core` issue_**

cc: @kdy1

**Reply by phoenisx** [reply]
Issue repro can be found here: https://github.com/phoenisx/swr-source-map-issue

**Reply by kdy1** [reply]
Sourcemap content being different is not an issue. Handling of `sources` can be the cause, but we need some investigation to fix it.

**Comment by sibelius** [comment]
is there a workaround for this?

**Reply by shay-alaluf** [reply]
try set 
    config.module.rules.push({
      test: /\.jsx?$/,
      exclude: /(node_modules)/,
      use: {
        loader: "swc-loader",
        options: {
          sourceMaps: false
        }
      }
    });
